### PR TITLE
Changes serialization of a pipeline in apps:info

### LINKED ISSUE #206: apps:info serializes a pipeline-coupling as the pipeline key
If you look at the serialization of `heroku apps:info APP --json`, you may think the serialization is confusing.

e.g.:

```
> heroku apps:info APP --json
   ...
  "pipeline": {
    "created_at": "2016-11-23T14:32:18+00:00",
    "app": {
      "id": APP_ID
    },
    "id": PIPELINE_COUPLING_ID,
    "pipeline": {
      "id": PIPELINE_ID,
      "name": "kolkrabbi"
    },
    "stage": "staging",
    "updated_at": "2016-11-23T14:32:18+00:00"
  }
}
```

Notice there is a pipeline nested in the main `pipeline` key. I personally think this is confusing, and I'd recommend changing the main `pipeline` key to `pipeline_coupling` instead.

### PR DESCRIPTION
**Pull request by raulb**
Fixes #206 

---

**Before this PR**

```
> heroku apps:info APP --json
   ...
  "pipeline": {
    "created_at": "2016-11-23T14:32:18+00:00",
    "app": {
      "id": APP_ID
    },
    "id": PIPELINE_COUPLING_ID,
    "pipeline": {
      "id": PIPELINE_ID,
      "name": "kolkrabbi"
    },
    "stage": "staging",
    "updated_at": "2016-11-23T14:32:18+00:00"
  }
}
```

**After this PR**

```
> heroku apps:info APP --json
   ...
  "pipeline_coupling": {
    "created_at": "2016-11-23T14:32:18+00:00",
    "app": {
      "id": APP_ID
    },
    "id": PIPELINE_COUPLING_ID,
    "pipeline": {
      "id": PIPELINE_ID,
      "name": "kolkrabbi"
    },
    "stage": "staging",
    "updated_at": "2016-11-23T14:32:18+00:00"
  }
}
```

cc @ys

### CODE REVIEW

File: test/commands/apps/info.js [hunk 1]
```diff
@@ -241,12 +241,17 @@ stack=cedar-14
...
+      .get('/apps/myapp/pipeline-couplings').reply(200, { app: { id: appAcm.id }, pipeline: { name: 'my-pipeline' } })
     return cmd.run({args: {app: 'myapp'}, flags: {json: true}})
       .then(() => expect(cli.stderr).to.equal(''))
       .then(() => {
         let json = JSON.parse(cli.stdout)
         expect(json.appExtended).to.equal(undefined)
         expect(json.app.extended).to.equal(undefined)
+        expect(json.addons.length).to.equal(addons.length)
+        expect(json.collaborators.length).to.equal(collaborators.length)
+        expect(json.dynos[0].type).to.equal('web')
```
Comment (raulb) [inline test/commands/apps/info.js]: These 3 extra lines is to check the existence of these other keys, which we weren't testing for before.

File: src/commands/apps/info.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ function * run (context, heroku) {
       }),
       dynos: heroku.get(`/apps/${app}/dynos`).catch(() => []),
       collaborators: heroku.get(`/apps/${app}/collaborators`).catch(() => []),
-      pipeline: heroku.get(`/apps/${app}/pipeline-couplings`).catch(() => null)
+      pipeline_coupling: heroku.get(`/apps/${app}/pipeline-couplings`).catch(() => null)
```
Comment (appleton) [inline src/commands/apps/info.js]: I don't know what the standard is in this repo, but I would've camel cased `pipelineCoupling` here.

Comment (raulb) [reply src/commands/apps/info.js]: @appleton I thought about that, but this is *also* to serialize it in the json output. Considered leaving it with the underscore to respect current format like `updated_at`. I could still convert it to underscore when rendering the json?

Comment (appleton) [reply src/commands/apps/info.js]: Oh that makes sense. I'd leave it alone in that case.

**Comment by jdxcode** [comment]
lgtm, but this would break for anyone currently depending on this behavior. I’d like for us to put a warning in place for a few weeks that this will be changing, maybe with a flag to move to the new behavior now.

I’m out for the next week but can look into doing this when I get back.

**Comment by jdxcode** [comment]
Unless someone else wants to take a stab at it of course

**Comment by raulb** [comment]
@jdxcode sounds good. I thought about that, but couldn't think of alternatives that were something different than rendering both keys `pipeline` and `pipeline_coupling`. I hadn't thought about using a flag. Do we have any precedent somewhere? i.e. have we used a particular flag with this purpose or should I come up with my own?

**Comment by jdxcode** [comment]
We don’t, but I’ve known this would happen eventually so we’ll have to come up with the pattern now. Maybe the flag could be —stage? Or maybe —new? —beta? I think I’m partial to —stage myself (similar to what es201x does)

And yeah, in this case I think we could output both with a warning without the flag, or output just pipeline_coupling with the flag

**Comment by raulb** [comment]
@jdxcode where would you include the warning though? I'm hesitant to include any warning for this particular case since it'd break any parsable operation that could be built in for command using `--json`.

Maybe we should include a similar model as we do in our [Platform API](https://devcenter.heroku.com/articles/api-compatibility-policy#deprecation):

> Deprecated resources will keep working for at least as long after deprecation as mandated by their stability: 1 month for prototype resources, 6 month for development resources and 12 months for production resources. Deprecated resources will not change stability.

**Comment by jdxcode** [comment]
I feel 12 months might be too long for a CLI change, I think 6 months might be more sensible. But yes, we should have a public statement just like API does.

You don’t need to worry about the warning though, the warning outputs to stderr so it won’t break any scripts. We also publicly state we never make any guarantees that stderr will be consistent (for this reason)

**Comment by raulb** [comment]
> You don’t need to worry about the warning though, the warning outputs to stderr so it won’t break any scripts. We also publicly state we never make any guarantees that stderr will be consistent (for this reason)

Oh, good point 👍

**Comment by raulb** [comment]
@jdxcode PR updated https://github.com/heroku/heroku-apps/pull/207/commits/a7d4c5ad6bcf3ebce1a0b1329aabf8555ed12f23.

**Pull request merged**